omegaconf/listconfig.py

```diff
@@ -500,20 +499,29 @@ def __iter__(self) -> Iterator[Any]:
 
     class ListIterator(Iterator[Any]):
         def __init__(self, lst: Any, resolve: bool) -> None:
-            self.iter = iter(lst.__dict__["_content"])
             self.resolve = resolve
+            self.iterator = iter(lst.__dict__["_content"])
             self.index = 0
+            from .base import Container
+            from .nodes import ValueNode
+
+            self.Container = Container
+            self.ValueNode = ValueNode
```

Comment:
The fact that this made a big difference in the perf is telling. We are importing in functions in a million places and it can actually be hurting performance in hotspots.